### PR TITLE
JIT: Ensure no overflow in ContainBlockStoreAddress

### LINKED ISSUE #76506: Test failure: access violation in cpblk tests
Affected tests:
- `JIT/Directed/PREFIX/volatile/1/cpblk/cpblk.sh`
- `JIT\\Directed\\PREFIX\\volatile\\1\\cpblk\\cpblk.cmd`

Runfo hits for [last 30 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E30+definition%3Aruntime+name%3A%22cpblk%22) as of 10/3:
- First occurrence on 9/30 in rolling run [36073](https://runfo.azurewebsites.net/view/build/?number=36073)
- Happened **2x-3x in each Rolling run** in last 3 days across various platforms (all are **arm64** -- on all platforms - Linux, OSX and Windows)

### Original Report

arm64, JitStress

Also fails in MinOpts

https://dev.azure.com/dnceng-public/public/_build/results?buildId=37829&view=ms.vss-test-web.build-test-results-tab&runId=754244&paneView=debug&resultId=108471

Started with 20220929.1 build:
https://dev.azure.com/dnceng-public/public/_build/results?buildId=37829&view=results

Tests:

JIT\\Directed\\PREFIX\\unaligned\\2\\cpblk\\cpblk.cmd
JIT\\IL_Conformance\\Old\\Conformance_Base\\ldc_c_cpblk\\ldc_c_cpblk.cmd
JIT\\Directed\\PREFIX\\unaligned\\4\\cpblk\\cpblk.cmd
JIT\\IL_Conformance\\Old\\Base\\cpblk\\cpblk.cmd
JIT\\IL_Conformance\\Old\\Conformance_Base\\c_cpblk\\c_cpblk.cmd
JIT\\IL_Conformance\\Old\\Conformance_Base\\cpblk\\cpblk.cmd
JIT\\Directed\\PREFIX\\unaligned\\1\\cpblk\\cpblk.cmd

Example:
```
  Starting:    JIT.Directed.XUnitWrapper (parallel test collections = on, max threads = 8)
    JIT\Directed\PREFIX\unaligned\4\cpblk\cpblk.cmd [FAIL]
      Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
         at _cpblk.main(System.String[])
```

@dotnet/jit-contrib 

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by jakobbotsch**
The offset here can be a "base" address due to various JIT transformations so we should ensure the range [offset, offset+size) does not overflow.

Fix #76506

### CODE REVIEW

**Comment by jakobbotsch** [comment]
Before #76273 we would see `GT_ADDEX` here so we wouldn't end up containing it.

File: src/coreclr/jit/lowerarmarch.cpp [hunk 1]
```diff
@@ -688,7 +688,12 @@ void Lowering::ContainBlockStoreAddress(GenTreeBlk* blkNode, unsigned size, GenT
     {
         return;
     }
-#endif // TARGET_ARM
+#else // !TARGET_ARM
+    if ((ClrSafeInt<int>(offset) + ClrSafeInt<int>(size)).IsOverflow())
```
Comment (jkotas) [inline src/coreclr/jit/lowerarmarch.cpp:692]: Can this overflow on arm too? Should this be outside the ifdef?

Comment (jakobbotsch) [reply src/coreclr/jit/lowerarmarch.cpp:692]: I think the ARM specific check above is fine -- we only get here for unrolled block copies so `size` is guaranteed to be small.

**Review by tannergooding — APPROVED** [review]

**Comment by tannergooding** [comment]
Some of the remaining failures may be related to: https://github.com/dotnet/runtime/pull/76517

**Comment by tannergooding** [comment]
The failure for Runtime_40607 is #76546

**Comment by jakobbotsch** [comment]
#76507 also still seems to be failing even with this fix.

**Comment by jakobbotsch** [comment]
Bruce just opened #76550 for the remaining failure.

**Comment by jakobbotsch** [comment]
superpmi-diffs failure is #76544. superpmi-replay failure is #76511. Still just waiting for the win-arm64 testing to finish...

**Pull request merged**